### ISSUE TITLE
Syncing app (Desktop and Mobile)

### ISSUE THREAD

**Issue by trymeouteh** [hudikhq/hoodik#105]
There is not a single FOSS cloud storage solution that supports E2EE and has good syncing clients on desktop and mobile.

I would like to purpose making a syncing app using Flutter. Flutter allows you to use one codebase and compile the app for Android, iOS, Windows, MacOS, and Linux.

It would be amazing to have a syncing client that supports the following...

- Multiple accounts
- Runs in the background (Desktop tray, mobile notification)
- Sync multiple folders in each account
- One way and two way sync
- Choose what folders sync over mobile data
- Ignore File (Like git ignore)
- Ignored subdirectories
- Intervals (Choose how often to do a sync)

**Comment by osousa** [comment]
I am using ente.io, it works quite well. 

Not perfect, but good.

**Comment by ngdangtu-vn** [comment]
May I suggest [Tauri](https://github.com/tauri-apps/tauri) framework? We could build a desktop + Android frontend from it. Moreover, it will fit better to the project since 50% code is already rust. Using Flutter framework will require dev learn Dart (it is a good, friendly to Typescript dev) and do we really many languages for the project?

**Comment by htunlogic** [comment]
A Flutter app with a Rust FFI bridge (reusing the encryption setup from the Rust crate) has been in development for some time already. The initial version is released on Android; iOS and macOS are coming soon(ish).

The app currently has basic functionality, but I will work on adding extra features that mobile/desktop development opens up.

**Comment by MintyDoggo** [comment]
> The initial version is released on Android; iOS and macOS are coming soon(ish).

Are there any plans for desktop linux and/or windows? This project is so cool and I'd love to be able to use it for syncing files on my linux desktop. Great work by the way!!

**Comment by 6oranges** [comment]
Related to this feature, it would be cool if the web app itself had local filesystem sync functionality as well. This could be accomplished for chromium based browsers via the [File System Access api](https://developer.chrome.com/docs/capabilities/web-apis/file-system-access). Though current caniuse support is [very low](https://caniuse.com/native-filesystem-api), as [firefox](https://github.com/mozilla/standards-positions/issues/154)/safari don't believe in users reading permission prompts ("[we do not think meaningful end user consent is possible to obtain](https://mozilla.github.io/standards-positions/#native-file-system)").

**Comment by htunlogic** [comment]
@MintyDoggo yes, there are plans for windows and linux app support, however it is challening right now due to inability to elegantly support the current markdown editor in there. It is a webview wrapper so that it can be the same experience on web and in-app. Rest of the app is Flutter with Rust FFI crypto bridge. And this kind of webview doesn't work on windows and linux. I am looking for a solution and will release the app as soon as possible.

@6oranges that's a great tip, I didn't realize that could be possible. I'll take a look at this and see what can be done to include it.
